src/tcg-to-deckbox.py

```diff
@@ -222,10 +218,34 @@ def getPathPrefix():
         ):
             row["Edition"] = "Mystery Booster Playtest Cards 2021"
 
+        ####################################################################
+        # Handle General Card Conversions
+        ####################################################################
+
+        # Remove all Parentheses at the end of cards
         row["Name"] = re.sub(r" \(.*\)", "", row["Name"])
         replace_strings(row, "NAMES", "Name")
-        if skip_scryfall_names == False and row["Name"] in scryfall_data:
-            row["Name"] = scryfall_data[row["Name"]]
+        # We need to do a little extra work on dual faced cards, because deckbox is inconsistent with whether it refers to cards by both names or just the front face.
+        if row["Name"] in scryfall_data:
+            deckbox_request_url = DECKBOX_URL + urllib.parse.quote(
+                scryfall_data[row["Name"]]
+            )
+            with requests.get(deckbox_request_url) as deckbox_response:
+
+                # If we are not redirected to a new page, then we should only use the front face name
+                if deckbox_request_url == deckbox_response.url:
```

Comment:
if deckbox_response.status_code == 302? 

Api seems to react to the user-agent header tho, as in if I use the base curl user-agent (experimentation located at https://gist.github.com/rkonell/27c093b4703cf2a7d7b354409fb1776b) it actually returns a 404, but if I provide it with any other user-agent then I get the expected 302.  But having to be user-agent aware to know what to expect seems brittle(is anything user-agent related NOT brittle?).

With that in mind however, we shouldn't bother even following the redirects and instead only test the status code of the response itself, by passing an optional parameter requests exposes:
```
with requests.get(deckbox_request_url, allow_redirect=False) as deckbox_response:
    if deckbox_response.status == 200:
    ...

Reply:
Unfortunately, need to be a little bit trickier than that. I tested with curl, so the user agent might change things, but I found out that valid cards redirect too, if they have multiple printings. It redirects to one of the printings. That's why I checked the response URL.

Reply:
oh interesting find...  what card if you don't mind?  I'd like to review the un-redirected response for clues.  I iterated through each dual-faced card to get the response for each and got 200s, 301s, and 302s, which the 301s i wasn't expecting.  I'm hoping maybe we can identify some response header that's shared amongst the variations (Found 0, Found 1, Found many).

Reply:
ugh, came across an odd issue with Lantern Bearer:
https://deckbox.org/mtg/Lantern%20Bearer%20/%20Lanterns%27%20Lift?printing=61146

If you notice... the name delimiter is a single slash, when normally it's two.  It actually redirects that way!  So when the comparison is performed, it doesn't match.  What's even better is if you try to upload a csv with the single slash version to match the url of the webpage, it rejects it.  It only accepts the double slash version..

Guessing they have some bug in the path between their webserver and the db that's dropping a slash.

Reply:
Oh I noticed also that the response url sometimes will contain a query parameter, That should be stripped off before comparing.

Reply:
(sorry if there was message spam here.. i had wrote out a comment and it displayed oddly on my end so i deleted)

Ran into an issue with Lantern Bearer: 
https://deckbox.org/mtg/Lantern%20Bearer%20/%20Lanterns%27%20Lift?printing=61146

url includes just a single slash as the name delimiter so it always fails comparison.  It's odd because it rejects a single slash when imported via csv, needs the double.  I'm guessing there is a defect between the webserver and their db.  I'm just manually adjusting for now.  

Another oddity I noticed with that card is it includes a query parameter in the response url.  Haven't had that issue with the other dual faced cards... but should strip query parameters from the response url prior to comparing in case there are others.

Reply:
That query parameter is exactly the issue I was talking about, I just hit it with older cards, such as Delver of Secrets, which has seen reprints. The single slash is annoying. I posted about the API issues but I haven't gotten a response yet. For now, I can check double slash, and if that fails, I can also check single slash